docs/design/permissive-mode.md

```diff
@@ -0,0 +1,353 @@
+
+Validation can be done using the SPDX [online](https://tools.spdx.org/app/validate)
+tool.
+
+#### Approach 1 (preferred): Annotations within the packagess
+
+This approach involves adding annotations directly to each package that describe
+permissive mode violations.
+
+**NOTE:** Annotations are currently used for converting CycloneDX component properties.
+
+Example:
+
+```json
+{
+  "spdxVersion": "SPDX-2.3",
+  "dataLicense": "CC0-1.0",
+  "SPDXID": "SPDXRef-DOCUMENT",
+  "name": "unknown",
+  "documentNamespace": "NOASSERTION",
+  "creationInfo": {
+    "licenseListVersion": "3.24",
+    "creators": ["Organization: Red Hat, Inc", "Tool: hermeto"],
+    "created": "2025-01-15T10:30:00Z"
+  },
+  "packages": [
+    {
+      "name": "github.com/gin-gonic/gin",
+      "downloadLocation": "NOASSERTION",
+      "SPDXID": "SPDXRef-Package-golang-github.com-gin-gonic-gin-1.9.1",
+      "versionInfo": "1.9.1",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:golang/github.com/gin-gonic/gin@v1.9.1"
+        }
+      ],
+      "annotations": [
+        {
+          "annotator": "Tool: hermeto",
+          "annotationDate": "2025-01-15T10:30:00Z",
+          "annotationType": "OTHER",
+          "comment": "hermeto:permissive-mode:violation:gomod:vendor-directory-changed-after-vendoring"
+        }
+      ]
+    },
+    {
+      "name": "github.com/gorilla/mux",
+      "downloadLocation": "NOASSERTION",
+      "SPDXID": "SPDXRef-Package-golang-github.com-gorilla-mux-1.8.0",
+      "versionInfo": "1.8.0",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:golang/github.com/gorilla/mux@v1.8.0"
+        }
+      ],
+      "annotations": [
+        {
+          "annotator": "Tool: hermeto",
+          "annotationDate": "2025-01-15T10:30:00Z",
+          "annotationType": "OTHER",
+          "comment": "hermeto:permissive-mode:violation:gomod:vendor-directory-changed-after-vendoring"
+        }
+      ]
+    }
+  ],
+  "relationships": [
+    {
+      "spdxElementId": "SPDXRef-DOCUMENT",
+      "relationshipType": "DESCRIBES",
+      "relatedSpdxElement": "SPDXRef-Package-golang-github.com-gin-gonic-gin-1.9.1"
+    },
+    {
+      "spdxElementId": "SPDXRef-DOCUMENT",
+      "relationshipType": "DESCRIBES",
+      "relatedSpdxElement": "SPDXRef-Package-golang-github.com-gorilla-mux-1.8.0"
+    }
+  ]
+}
+```
+
+#### Approach 2 (alternative): Top-level annotations with package references
+
+This approach involves adding top-level annotations that describe permissive mode
+violations and referencing the affected components by their IDs. Similar to the
+CycloneDX approach 1.
+
+However, only since SPDX version 3, annotations can become actual SPDX elements
```

Comment:
This is irrelevant to approach with a list of SPDXIDs in a well-defined annotation. To keep this paragraph I think you would need to at least add something like this: "SPDX spec allows defining relationships between SPDX elements of multitude of types. Two `spdxElementId` are required for that one per element in a relationship.". Maybe this should be moved to `Approach 2.5` section?